pkg/virt-launcher/virtwrap/sriov/hostdev.go

```diff
@@ -98,3 +102,82 @@ func decoratePciAddressField(addressField string) (*api.Address, error) {
 	}
 	return decoratedAddrField, nil
 }
+
+type deviceDetacher interface {
+	DetachDevice(xmlData string) error
+}
+
+func DetachHostDevices(dom deviceDetacher, hostDevices []api.HostDevice) error {
+	for _, device := range hostDevices {
+		devXML, err := xml.Marshal(device)
+		if err != nil {
+			return fmt.Errorf("failed to encode (xml) device %v, err: %v", device, err)
+		}
+		err = dom.DetachDevice(string(devXML))
+		if err != nil {
+			return fmt.Errorf("failed to detach device %s, err: %v", devXML, err)
+		}
+		log.Log.Infof("Live migration, hot-unplug device: %s (%v)", device.Alias.Name, device.Source.Address)
```

Comment:
I think the "Live migration" part of this message should be logged at the caller of this function,
at this abstraction level we shouldn't mention why we unplug a HostDevice